mailbox/backup/src/test/java/org/apache/james/mailbox/backup/ZipperTest.java

```diff
@@ -221,6 +237,19 @@ void archiveShouldWriteMailboxAnnotationsMetadataWhenOneThousandPresent() throws
         }
     }
 
+    @Test
+    void archiveShouldThrowWhenAnnotationExtraFieldLengthTooLarge() throws Exception {
+        testee.archive(ImmutableList.of(new MailboxWithAnnotations(MAILBOX_1, tenThousandAnnotations)), Stream.of(), output);
+
+        try (ZipFile zipFile = new ZipFile(toSeekableByteChannel(output))) {
+            assertThatThrownBy(() -> assertThatZip(zipFile)
+                .containsOnlyEntriesMatching(
+                    hasName(MAILBOX_1.getName() + "/")
+                        .containsExtraFields(new MailBoxAnnotationsExtraField(tenThousandAnnotations)))
+            ).isInstanceOf(IllegalArgumentException.class);
```

Comment:
and now the big question: do we really want to throw (ie fail an export) if annotation is too long, or just issue a warning?

Reply:
With the refactoring to store the annotations as entries. This test will disappear